Demo/Demo/ViewModels/CardVaultViewModel.swift

```diff
@@ -0,0 +1,82 @@
+import UIKit
+import CardPayments
+import CorePayments
+
+class CardVaultViewModel: ObservableObject, CardVaultDelegate {
+
+    @Published var state = CardVaultState()
+
+    func getSetupToken(
+        customerID: String? = nil,
+        selectedMerchantIntegration: MerchantIntegration
+    ) async throws {
+        do {
+            let setupTokenResult = try await DemoMerchantAPI.sharedService.getSetupToken(
+                customerID: customerID,
+                selectedMerchantIntegration: selectedMerchantIntegration
+            )
+            DispatchQueue.main.async {
+                self.state.setupTokenResponse = setupTokenResult
+            }
+        } catch {
+            throw error
+        }
+    }
+
+    func getPaymentToken(
+        setupToken: String,
+        selectedMerchantIntegration: MerchantIntegration
+    ) async throws {
+        do {
+            let paymentTokenResult = try await DemoMerchantAPI.sharedService.getPaymentToken(
+                setupToken: setupToken,
+                selectedMerchantIntegration: selectedMerchantIntegration
+            )
+            DispatchQueue.main.async {
+                self.state.paymentTokenResponse = paymentTokenResult
+            }
+        } catch {
+            throw error
+        }
+    }
+
+    func vault(
+        config: CoreConfig,
+        card: Card,
+        setupToken: String
+    ) async {
+        Task {
+            let cardClient = CardClient(config: config)
+            cardClient.vaultDelegate = self
+
+            let cardVaultRequest = CardVaultRequest(card: card, setupTokenID: setupToken)
+            cardClient.vault(vaultRequest: cardVaultRequest)
+        }
+    }
+
+    func isCardFormValid(cardNumber: String, expirationDate: String, cvv: String) -> Bool {
+        let cleanedCardNumber = cardNumber.replacingOccurrences(of: " ", with: "")
+        let cleanedExpirationDate = expirationDate.replacingOccurrences(of: " / ", with: "")
+
+        let enabled = cleanedCardNumber.count >= 15 && cleanedCardNumber.count <= 19
+        && cleanedExpirationDate.count == 4 && cvv.count >= 3 && cvv.count <= 4
+        return enabled
+    }
+
+    func updateUpdateSetupTokenResult(vaultResult: CardPayments.CardVaultResult) {
+        DispatchQueue.main.async {
```

Comment:
This could probably use a better name. The viewModel state needs to be updated so the SwiftUI view gets updated with result and only way we get the "done" status from the SDK is through the delegate function.

Reply:
On Android we use a Kotlin feature similar to Swift's computed properties, that could be a good fit here as well.